graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/RequiredTypeCollector.kt

```diff
@@ -34,22 +34,26 @@ import graphql.util.TraverserContext
 
 class RequiredTypeCollector(
     private val document: Document,
-    queries: Set<String> = emptySet(),
-    mutations: Set<String> = emptySet(),
-    subscriptions: Set<String> = emptySet()
+    config: CodeGenConfig
 ) {
     val requiredTypes: Set<String> = LinkedHashSet()
 
     init {
         val fieldDefinitions = mutableListOf<FieldDefinition>()
         for (definition in document.definitions.asSequence().filterIsInstance<ObjectTypeDefinition>()) {
             when (definition.name) {
-                "Query" -> definition.fieldDefinitions.filterTo(fieldDefinitions) { it.name in queries }
-                "Mutation" -> definition.fieldDefinitions.filterTo(fieldDefinitions) { it.name in mutations }
-                "Subscription" -> definition.fieldDefinitions.filterTo(fieldDefinitions) { it.name in subscriptions }
+                "Query" -> definition.fieldDefinitions.filterTo(fieldDefinitions) { it.name in config.includeQueries }
+                "Mutation" -> definition.fieldDefinitions.filterTo(fieldDefinitions) { it.name in config.includeMutations }
+                "Subscription" -> definition.fieldDefinitions.filterTo(fieldDefinitions) { it.name in config.includeSubscriptions }
+                else -> {
+                    if (config?.generateClientApiv2 == true || config?.generateClientApi == true) {
+                        fieldDefinitions.addAll(definition.fieldDefinitions)
```

Comment:
This is adding all definitions , right? Not just input types.

Reply:
Correct, while I noticed an input type was missing, other types may be implicated and here we are telling the node visitor to look at all the definitions for possible input types, or other requireable types